lesson23/js/components/table.js

```diff
@@ -0,0 +1,360 @@
+      thFragment.appendChild(th);
+    });
+    thead.appendChild(tr).appendChild(thFragment);
+    return thead;
+  },
+
+  getTableBody(membersMap) {
+    const tbody = createAttributedElements("tbody", {
+      id: "tbody",
+      class: "tbody",
+    });
+    tbody.appendChild(this.getTrFragment(membersMap));
+    return tbody;
+  },
+
+  getTrFragment(membersMap) {
+    const trFragment = document.createDocumentFragment();
+    Object.values(membersMap).forEach((memberMap) => {
+      const tr = createAttributedElements("tr", { class: "tr" });
+      trFragment.appendChild(tr).appendChild(this.getTdFragment(memberMap));
+    });
+    return trFragment;
+  },
+
+  getTdFragment(memberMap) {
+    const tdFragment = document.createDocumentFragment();
+    Object.keys(tableConfig.column).forEach((key) => {
+      if (key === "sex") {
+        for (const [sexType, sexTypeValue] of Object.entries(
+          tableConfig.column.sex.type
+        )) {
+          if (memberMap[key] === sexType) {
+            const td = createAttributedElements(
+              "td",
+              { class: "td" },
+              sexTypeValue.japanese
+            );
+            tdFragment.appendChild(td);
+          }
+        }
+        return;
+      }
+      const td = createAttributedElements(
+        "td",
+        { class: "td" },
+        memberMap[key]
+      );
+      tdFragment.appendChild(td);
+    });
+    return tdFragment;
+  },
+
+  getsortButton(key) {
+    const sortButton = createAttributedElements("button", {
+      id: key,
+      class: "button",
+      "data-sort": `${tableConfig.Sort.Both}`,
+    });
+    sortButton.addEventListener("click", clickSortButton);
+    return sortButton;
+  },
+};
+
+const paginationParts = {
+  renderPaginationButtonContainer(currentPage, totalPage) {
+    const paginationButtonContainer = createAttributedElements("div", {
+      class: "pagination-button__container",
+    });
+    tableConfig.parent.appendChild(paginationButtonContainer);
+    paginationButtonContainer.appendChild(this.getDirectionButtonsFragment());
+    paginationButtonContainer.appendChild(
+      this.getCurrentPageTextElement(currentPage, totalPage)
+    );
+  },
+
+  renderPaginationTextElement(currentPage, totalPage) {
+    const paginationTextElement = createAttributedElements(
+      "p",
+      {
+        id: "js-paginationText",
+        class: "pagination-text",
+      },
+      `${(currentPage + 1) * RowsCount}件/${totalPage * RowsCount}件中`
+    );
+    tableConfig.parent.appendChild(paginationTextElement);
+  },
+
+  getDirectionButtonsFragment() {
+    const directionButtonsFragment = document.createDocumentFragment();
+    Object.keys(directions).forEach((direction) => {
+      const button = createAttributedElements(
+        "button",
+        {
+          id: `js-pagination-${directions[direction].value}-button`,
+          class: `pagination-button _${directions[direction].value}`,
```

Comment:
nits
It seems better to remove unnecessary spaces.
```suggestion
          class: `pagination-button_${directions[direction].value}`,
```

Reply:
The space in the part you commented on is what I dared to leave blank. Sorry for the confusion 😭 (I did so because it will be useful when I apply css in the future! I should have written it in the PR 🙇)

I was assuming that the css for the common areas would be defined first, and then the styles for next and prev would be applied.

Reply:
I'm sorry I misunderstood.
Ignore this nits.
 It's useful when adding class names!

Reply:
Thank you! I removed `_`as it is true that I also had a naming scheme that was difficult to understand.
[1bf88c4](https://github.com/443n0511/js-lesson/pull/45/commits/1bf88c46a14999055991746520417084ae200c14)

I will try to make the naming easy to understand for everyone to see. Thanks for the helpful comments.